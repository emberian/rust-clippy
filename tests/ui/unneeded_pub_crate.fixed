// run-rustfix
#![allow(dead_code)]
#![warn(clippy::unneeded_pub_crate)]

 struct Baz {
     field: u8,
}

mod outer {
    mod inner {
        // do these things _really_ need to be `pub(crate)`?
         struct Foo;
         trait Baz {
            fn the_goods(&self);
            fn secret_thing(&self);
        }
        impl Baz for crate::Baz {
            fn the_goods(&self) {}
            fn secret_thing(&self) {}
        }
         fn foo() {}
        pub(crate) fn bar() -> super::ReturnStruct {
            foo();
            let _ = Foo;
            let x = crate::Baz { field: 3 };
            x.the_goods();
            x.secret_thing();
            super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            }
        }
        pub(crate) struct S1(u8, pub(crate) u8);
    }
    pub(crate) struct ReturnStruct {
        pub(crate) used_outside: u8,
         not_used_outside: u8,
    }
    pub(crate) fn main() -> (ReturnStruct, inner::S1) {
        inner::bar()
    }
}

fn main() {
    let (value_outside, s1) = crate::outer::main();
    s1.1 = 42;
    let _ = value_outside.used_outside;
}
