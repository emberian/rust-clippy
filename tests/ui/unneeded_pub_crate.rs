// run-rustfix
#![allow(dead_code)]
#![warn(clippy::unneeded_pub_crate)]

pub(crate) struct Baz {
    pub(crate) field: u8,
}

mod outer {
    mod inner {
        // do these things _really_ need to be `pub(crate)`?
        pub(crate) struct Foo;
        pub(crate) trait Baz {
            fn the_goods(&self);
            fn secret_thing(&self);
        }
        impl Baz for crate::Baz {
            fn the_goods(&self) {}
            fn secret_thing(&self) {}
        }
        pub(crate) fn foo() {}
        pub(crate) fn bar() -> super::ReturnStruct {
            foo();
            let _ = Foo;
            let x = crate::Baz { field: 3 };
            x.the_goods();
            x.secret_thing();
            super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            }
        }
    }
    pub(crate) struct ReturnStruct {
        pub(crate) used_outside: u8,
        pub(crate) not_used_outside: u8,
    }
    pub(crate) fn main() -> ReturnStruct {
        inner::bar()
    }
}

fn main() {
    let value_outside = crate::outer::main();
    let _ = value_outside.used_outside;
}
