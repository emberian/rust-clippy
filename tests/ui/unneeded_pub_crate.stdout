about to look for:
HirId { owner: DefId(0:3 ~ unneeded_pub_crate[317d]::Baz), local_id: 3 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:0 ~ unneeded_pub_crate[317d]), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:6:5: 6:15 (#0), linked_fate: [] }
HirId { owner: DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:5 ~ unneeded_pub_crate[317d]::outer), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:34:5: 34:15 (#0), linked_fate: [] }
HirId { owner: DefId(0:7 ~ unneeded_pub_crate[317d]::outer::inner::Foo), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:6 ~ unneeded_pub_crate[317d]::outer::inner), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:12:9: 12:19 (#0), linked_fate: [HirId { owner: DefId(0:7 ~ unneeded_pub_crate[317d]::outer::inner::Foo), local_id: 1 }] }
HirId { owner: DefId(0:3 ~ unneeded_pub_crate[317d]::Baz), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:0 ~ unneeded_pub_crate[317d]), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:5:1: 5:11 (#0), linked_fate: [] }
HirId { owner: DefId(0:9 ~ unneeded_pub_crate[317d]::outer::inner::Baz), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:6 ~ unneeded_pub_crate[317d]::outer::inner), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:13:9: 13:19 (#0), linked_fate: [] }
HirId { owner: DefId(0:15 ~ unneeded_pub_crate[317d]::outer::inner::foo), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:6 ~ unneeded_pub_crate[317d]::outer::inner), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:21:9: 21:19 (#0), linked_fate: [] }
HirId { owner: DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct), local_id: 3 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:5 ~ unneeded_pub_crate[317d]::outer), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:35:9: 35:19 (#0), linked_fate: [] }
HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 0 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:6 ~ unneeded_pub_crate[317d]::outer::inner), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:22:9: 22:19 (#0), linked_fate: [] }
HirId { owner: DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct), local_id: 6 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:5 ~ unneeded_pub_crate[317d]::outer), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:36:9: 36:19 (#0), linked_fate: [] }
HirId { owner: DefId(0:7 ~ unneeded_pub_crate[317d]::outer::inner::Foo), local_id: 1 } \t => WatchedItem { enclosing_module: HirId { owner: DefId(0:6 ~ unneeded_pub_crate[317d]::outer::inner), local_id: 0 }, status: Unreferenced, span: $DIR/unneeded_pub_crate.rs:12:9: 12:19 (#0), linked_fate: [HirId { owner: DefId(0:7 ~ unneeded_pub_crate[317d]::outer::inner::Foo), local_id: 0 }] }
DefId(0:8 ~ unneeded_pub_crate[317d]::outer::inner::Foo::{constructor#0}) used in struct outer::inner::Foo (hir_id=HirId { owner: DefId(0:7 ~ unneeded_pub_crate[317d]::outer::inner::Foo), local_id: 0 }); it's a local reference
DefId(0:9 ~ unneeded_pub_crate[317d]::outer::inner::Baz) used in unknown node (hir_id=HirId { owner: DefId(0:0 ~ unneeded_pub_crate[317d]), local_id: 0 }); it's a local reference
DefId(0:3 ~ unneeded_pub_crate[317d]::Baz) used in type crate::Baz (hir_id=HirId { owner: DefId(0:12 ~ unneeded_pub_crate[317d]::outer::inner::{impl#0}), local_id: 3 }); it's a local reference
DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct) used in type super::ReturnStruct (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 37 }); it's a local reference
DefId(0:15 ~ unneeded_pub_crate[317d]::outer::inner::foo) used in expr foo (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 4 }); it's a local reference
DefId(0:8 ~ unneeded_pub_crate[317d]::outer::inner::Foo::{constructor#0}) used in expr Foo (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 8 }); it's a local reference
DefId(0:3 ~ unneeded_pub_crate[317d]::Baz) used in expr crate::Baz { field: 3 } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 15 }); it's a local reference
DefId(0:4 ~ unneeded_pub_crate[317d]::Baz::field) used in expr crate::Baz { field: 3 } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 15 }); it's a local reference
DefId(0:3 ~ unneeded_pub_crate[317d]::Baz) used in expr crate::Baz { field: 3 } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 15 }); it's a local reference
DefId(0:10 ~ unneeded_pub_crate[317d]::outer::inner::Baz::the_goods) used in expr x.the_goods() (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 21 }); it's not tracked
DefId(0:11 ~ unneeded_pub_crate[317d]::outer::inner::Baz::secret_thing) used in expr x.secret_thing() (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 26 }); it's not tracked
DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct) used in expr super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 34 }); it's a local reference
DefId(0:18 ~ unneeded_pub_crate[317d]::outer::ReturnStruct::used_outside) used in expr super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 34 }); it's a local reference
DefId(0:19 ~ unneeded_pub_crate[317d]::outer::ReturnStruct::not_used_outside) used in expr super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 34 }); it's a local reference
DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct) used in expr super::ReturnStruct {
                used_outside: 0,
                not_used_outside: 0,
            } (hir_id=HirId { owner: DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar), local_id: 34 }); it's a local reference
DefId(0:17 ~ unneeded_pub_crate[317d]::outer::ReturnStruct) used in type ReturnStruct (hir_id=HirId { owner: DefId(0:20 ~ unneeded_pub_crate[317d]::outer::main), local_id: 7 }); it's a local reference
DefId(0:16 ~ unneeded_pub_crate[317d]::outer::inner::bar) used in expr inner::bar (hir_id=HirId { owner: DefId(0:20 ~ unneeded_pub_crate[317d]::outer::main), local_id: 3 }); it's a crate reference
DefId(0:20 ~ unneeded_pub_crate[317d]::outer::main) used in expr crate::outer::main (hir_id=HirId { owner: DefId(0:21 ~ unneeded_pub_crate[317d]::main), local_id: 6 }); it's not tracked
